firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -169,6 +178,16 @@ private static boolean persistedFidMissingOrInErrorState(PersistedFidEntry persi
     };
   }
 
+  @NonNull
+  private static <F, T> Continuation<F, T> call(@NonNull Supplier<T> supplier) {
+    return t -> {
+      if (!t.isComplete()) {
+        Tasks.await(t);
```

Comment:
@ciarand  Instrumented tests fail sometimes because of this line. Its not strictly waiting for the prev Task (getId) to complete. Ends up calling refreshAuthTokenIfNecessary with empty PersistedFidEntry ( pointed you to that code). 

Can you please help me to understand what am I missing?

Reply:
Test failure is alternating.

Reply:
I don't think this is correct. You're calling `Tasks.call(executor, this::getId)` but `getId` is already calling `Tasks.call` so you're doubly nesting your async operations.

Instead of this:

```java
    return Tasks.call(executor, this::getId)
        .continueWith(call(() -> refreshAuthTokenIfNecessary(forceRefresh)));
```

I think you mean to do something like this:

```java
    return getId().continueWith((idTask) -> refreshAuthTokenIfNecessary(idTask.getResult(), forceRefresh));
```

NOTE: The call to `getResult` will throw an Exception if the `getId` task failed, which will cause the return value of this method (the `getAuthToken` method) to be a failed Task, which is afaict what we want.

You're also not passing an explicit executor into the .continueWith call though, so the refreshAuthTokenIfNecessary method is getting run on the main thread (!). Given you're not allowed to call Tasks.await on the main thread, that seems like a bug.

Reply:
Very good catch. You are right. That was the issue :) Fixed it. About the 'Tasks.await' it wasn't a bug until I called ```Tasks.call(executor, this::getId)``` . After the change recommended by you, I could not await in the Main Thread. I updated the  code to await using the ```executor```. Let me know what you think.